CTFd/models/__init__.py

```diff
@@ -244,7 +244,10 @@ class Users(db.Model):
     verified = db.Column(db.Boolean, default=False)
 
     # Relationship for Teams
-    team_id = db.Column(db.Integer, db.ForeignKey("teams.id"))
+    # use_alter=True along with name='' adds this foreign key after Teams has been created to avoid circular dependency
+    team_id = db.Column(
+        db.Integer, db.ForeignKey("teams.id", use_alter=True, name="users_team_id_fkey")
```

Comment:
I don't like pinning the name here. If we can tap into the Sqlalchemy built-in name generator, I would be more willing to merge this in but I still am not certain I want to go down the path of supporting multiple databases overall.

Reply:
Removed the name and let sqlalchemy to set the name of the constraint using its internal conventions: https://alembic.sqlalchemy.org/en/latest/naming.html